### PR TITLE
 LOVE-638 add docker-compose files for official blueprints

### PR DESCRIPTION
**Pull request by deepu105**
This needs to be cherrypicked to master and released as well

## Definition of Done

This is the definition of done for (new and modified) blueprints to be accepted into the [central XebiaLabs blueprints repository](https://github.com/xebialabs/blueprints).

### Value

This section describes how to determine whether a blueprint has enough value to be included in the central XebiaLabs blueprints repository.

- [x] The blueprint applies to a focused use case.
- [x] The blueprint uses at least one XebiaLabs product.
- [x] The blueprint depends on as few other products as possible. **N.B.:** This is part of the definition of done to keep the blueprint focused and applicable to as many users as possible. A blueprint that depends on one specific CI tool **and** one specific Java EE application server **and** one specific ticketing system can only be used by users that have all three products.
- [x] The blueprint can be used with content supplied by the user but also includes demo content. A blueprint that does not work with content supplied by the user is a demo or a showcase and does not help the user with their own work.

### Technical

This section describes how to determine whether the blueprint has the right technical quality to be included in the central XebiaLabs blueprints repository.

- [x] The branch from which the PR is created is up-to-date with the `development` branch.
- [x] The blueprint contains a `README.md` file at the root of the blueprint folder that describes the blueprint, using the README template in the `.github` folder of this repository.
- [x] The blueprint generates a `xebialabs/USAGE.md` file which, at a minimum, explains how to use the blueprint after it is generated (like adding any missing steps, creating accounts, setting up Docker containers, applying the YAML using `xl` CLI, running release. etc.). **N.B.:** Do not use this document to describe how to instantiate the blueprint. It will only be available to the user *after* the blueprint has been instantiated.
- [x] The blueprint generates a `docker/docker-compose.yml` file which can be used to spin up the required non-XebiaLabs products that the blueprints uses. This can be used to test and/or try out the blueprint without having to manually install those products.
- [x] The blueprint does not contain sensitive information such passwords, tokens, credentials or licenses.
- [x] The blueprint uses `secret: true` in the `blueprint.yaml` parameter definition for question that ask for sensitive information.
- [x] The blueprint does not contain the files `xebialabs/.gitignore`, `xebialabs/values.xlvals` and `xebialabs/secrets.xlvals` and does not refer to them from the files section of the `blueprint.yaml` file. These files will be generated when the blueprint is instantiated. To generate the `xebialabs/values.xlvals` file, use the `saveInXlVals: true` directive in the parameters section of the blueprint. To generate the `xebialabs/secrets.xlvals` file, use the `secret: true` directive .
- [x] The blueprint does not define parameters for trivial things like phase names, task names, folder names etc. Ask questions that add value and be opinionated where possible. For example, ask for a project name and derive folder names, task names etc from that.
- [x] Folder and file names must use [kebab-case](http://wiki.c2.com/?KebabCase), for example: `aws/sample-app-demo`, `xld-environment.yaml`

### Review and testing

This section describes how to determine whether the blueprint has been reviewed and tested well enough to be included in the central XebiaLabs blueprints repository.

- [x] Unit test is added in a `__test__` folder for each blueprint. Refer the `CONTRIBUTING.md` file at the root of of this repository for more details.
- [x] The Travis CI for the PR is green
- [x] The blueprint has been reviewed by someone else in the team.
- [x] Both README and USAGE files have been reviewed by a technical writer and a product marketing manager.
- [x] The blueprint has been manually tested with the `docker/docker-compose.yml` that is generated as part of it.
- [x] The blueprint works on Linux, Mac and Windows.

### CODE REVIEW

File: docker/simple-demo-app/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  xl-cli:
+    image: xebialabsunsupported/xl-cli:8.6.1
+    depends_on:
+    - xl-deploy
+    - xl-release
+    command: ["apply", "--xl-deploy-url", "http://xl-deploy:4516/", "--xl-release-url", "http://xl-release:5516/", "-f", "/data/configure-xl-devops-platform.yaml"]
+    volumes:
+    - ./data/:/data:ro
+
+  dockerproxy:
```
Comment (deepu105) [inline docker/simple-demo-app/docker/docker-compose.yml]: @vpartington you mentioned that you found a cleaner way to do this, can we use it here as well?

File: docker/simple-demo-app/README.md [hunk 2]
```diff
@@ -58,7 +58,6 @@ This blueprint requires:
 
 This blueprint will output:
 
-* Sample Docker images
```
Comment (jkrieger-xl) [inline docker/simple-demo-app/README.md]: Maybe add something like `Sample docker-compose file`

File: aws/microservice-ecommerce/xebialabs/USAGE.md.tmpl [hunk 3]
```diff
@@ -8,16 +8,23 @@ You can check a detailed tutorial for this blueprint [here](https://docs.xebiala
...
+Please note that you can obtain a trial license or bring your own license for XL Release and XL Deploy if you like. Follow the steps below to use the Docker Compose files.
+
+    1. CD into the folder "docker" generated by the blueprint.
+    2. If you are using Jenkins for CI/CD please follow these additional steps
+        1. Set the following environment variables before starting the docker containers
+
+        ```
+        export GITHUB_USER=<your GitHub user name>
+        export DOCKER_USER=<your Docker user name>
+        export DOCKER_PASS=<your Docker password>
```
Comment (jkrieger-xl) [inline aws/microservice-ecommerce/xebialabs/USAGE.md.tmpl]: Missing `GITHUB_TOKEN`

File: aws/monolith-terraform/xebialabs/USAGE.md.tmpl [hunk 4]
```diff
@@ -8,15 +8,11 @@ The application is available in the repo [https://github.com/xebialabs/e-commerc
 
 ### Prerequisites
 
-If you do not have XL Release and XL Deploy instances running, you can use this Docker Compose setup to spin them up. The setup also contains a preconfigured Jenkins instance.
-Please note that you need to obtain a trial license or bring your own license for XL Release and XL Deploy. Follow the steps below to use the Docker Compose files.
+If you do not have XL Release and XL Deploy instances running, you can use the Docker Compose setup generated by the blueprint to spin them up.
+Please note that you can obtain a trial license or bring your own license for XL Release and XL Deploy if you like. Follow the steps below to use the Docker Compose files.
```
Comment (jkrieger-xl) [inline aws/monolith-terraform/xebialabs/USAGE.md.tmpl]: Maybe we could explain how to apply automatically their own license? They can use the `default-conf` folder right?

File: aws/microservice-ecommerce/xebialabs/USAGE.md.tmpl (hunk 3 above)
Comment (deepu105) [reply aws/microservice-ecommerce/xebialabs/USAGE.md.tmpl]: its needed only for private repo, but ya i'll add it

File: aws/monolith-terraform/xebialabs/USAGE.md.tmpl (hunk 4 above)
Comment (deepu105) [reply aws/monolith-terraform/xebialabs/USAGE.md.tmpl]: not anymore, seems like you need to set them in env variable and stuff. Maybe I can point to some docs on how to add licence

File: docker/simple-demo-app/README.md (hunk 2 above)
Comment (deepu105) [reply docker/simple-demo-app/README.md]: ok

Comment (deepu105) [reply docker/simple-demo-app/README.md]: done

File: aws/monolith-terraform/xebialabs/USAGE.md.tmpl (hunk 4 above)
Comment (deepu105) [reply aws/monolith-terraform/xebialabs/USAGE.md.tmpl]: seems like there is no clear docs on licence installtion for docker and with Jetpack and stuff its even more confusing so lets leave it out here

Comment (jkrieger-xl) [reply aws/monolith-terraform/xebialabs/USAGE.md.tmpl]: shall we merge, then?

Comment (deepu105) [reply aws/monolith-terraform/xebialabs/USAGE.md.tmpl]: The jetpack pages also doesnt say about licence. Anyways I think when licence expires you can add it via UI

**Comment by deepu105** [comment]
Thanks, I have merged to master as well and published this

**Pull request merged**